src/MainWindow.vala

```diff
@@ -331,7 +331,7 @@ namespace Terminal {
         protected bool match_keycode (int keyval, uint code) {
 #endif
             Gdk.KeymapKey [] keys;
-            Gdk.Keymap keymap = Gdk.Keymap.get_default ();
+            Gdk.Keymap keymap = Gdk.Keymap.get_for_display (get_display ());
```

Comment:
Might as well follow current style and use `var keymap = ...` here.

Reply:
done